zio-http/jvm/src/main/scala/zio/http/netty/NettyBody.scala

```diff
@@ -166,7 +174,7 @@ object NettyBody extends BodyEncoding {
               maybeError =>
                 ZChannel.fromZIO(queue.shutdown) *>
                   maybeError.fold[ZChannel[Any, Any, Any, Any, E, Chunk[A], Unit]](ZChannel.unit)(ZChannel.fail(_)),
-              a => ZChannel.write(a) *> loop,
+              a => ZChannel.write(a) *> ZChannel.fromZIO(nettyRead.whenZIO(queue.isEmpty)) *> loop,
```

Comment:
I'm wondering whether we should buffer reads here, e.g., `nettyRead.whenZIO(queue.size.map(_ < XXX))`

Previously this was set to 4096 (using a bounded queue which we can't use anymore since it was blocking Netty's threads) and there weren't any issues right?

Reply:
Also, 2 notes on performance. 
1. Use `whenZIODiscard` instead of `whenZIO`
2. I think it might be better to store this ZChannel in a val to avoid creating it on each loop:

```scala
val maybeRead = ZChannel.fromZIO(nettyRead.whenZIODiscard(???))

// And then
a => ZChannel.write(a) *> maybeRead *> loop,
```

Reply:
I'll fix the suggested changes in NettyBody!